Adding python changes for numpy reader

SundarRajan28 opened the pull request.

SundarRajan28 committed:
Adding python changes for numpy reader

fiona-gladwin commented on rocAL_pybind/amd/rocal/plugin/generic.py:
```diff
@@ -218,6 +218,57 @@ def __init__(self,
+        if (self.last_batch_policy is (types.LAST_BATCH_PARTIAL)) and b.getRemainingImages(self.loader._handle) < self.batch_size:
+            if (self.last_batch_size is None):
+                self.last_batch_size = self.batch_size - \
+                    b.getLastBatchPaddedSize(self.loader._handle)
```
Move to prev line itself

SundarRajan28 replied:
This is autopep8 formatting

fiona-gladwin commented on rocAL_pybind/amd/rocal/plugin/generic.py:
```diff
@@ -218,6 +218,57 @@ def __init__(self,
                                                           multiplier=pipe._multiplier, offset=pipe._offset, display=display, device=device, device_id=device_id)
 
 
+class ROCALNumpyIterator(object):
```
Do we need a separate iterator, why not use the generic iterator for this
And we can check if it is numpy reader and only return the output list

In generic iterator only the last batch check is missing I feel that can be added please check

SundarRajan28 replied:
In numpy reader, there is no labels so we return only images. In generic iterator, even though label reader is not present, it returns label tensor which is set as None

fiona-gladwin replied:
But we can have if conditions to check if it is Numpy reader right?
I feel that the use of another iterator adds extra code

fiona-gladwin commented on rocAL_pybind/amd/rocal/plugin/pytorch.py:
```diff
@@ -401,6 +401,68 @@ def __len__(self):
         b.rocalRelease(self.loader._handle)
 
+
+class ROCALNumpyIterator(object):
```
Same as generic.py file comment
Why don't we use the existing generic iterator with if condition for Numpy loader as required

SundarRajan28 replied:
In numpy reader, there is no labels so we return only images. In generic iterator, even though label reader is not present, it calls get_image_labels in the else case

fiona-gladwin commented on rocAL_pybind/amd/rocal/readers.py:
```diff
@@ -382,3 +382,18 @@ def webdataset(path, index_paths="", ext = None, missing_components_behavior = t
     webdata_metadata = b.webDatasetReader(
         Pipeline._current_pipeline._handle, *(kwargs_pybind.values()))
     return webdata_metadata
+
```
Remove one line

fiona-gladwin commented on rocAL_pybind/amd/rocal/types.py:
```diff
@@ -58,6 +59,7 @@
 from rocal_pybind.types import RGB_PLANAR
 
 #     RocalTensorLayout
+from rocal_pybind.types import NONE
```
Are we using NONE layout anywhere?
If not please don't add it as it was already a concern raised

SundarRajan28 replied:
None is used as the default layout in case user doesn't specify a layout. It is used in cosmoflow where the labels shape is (4,) and doesn't belong to any layout

fiona-gladwin commented on rocAL_pybind/amd/rocal/types.py:
```diff
@@ -147,6 +150,7 @@
     MAX_SIZE_ORIG: ("MAX_SIZE_ORIG", MAX_SIZE_ORIG),
     USER_GIVEN_SIZE_ORIG: ("USER_GIVEN_SIZE_ORIG", USER_GIVEN_SIZE_ORIG),
 
+    NONE: ("NONE", NONE),
```
Is it used?

SundarRajan28 replied:
Used as the default layout in numpy reader function if user doesn't specify a layout

fiona-gladwin replied:
Do we have equivalent type in C++?

fiona-gladwin commented on rocAL_pybind/amd/rocal/types.py:
```diff
@@ -38,6 +38,7 @@
 from rocal_pybind.types import FLOAT
 from rocal_pybind.types import FLOAT16
 from rocal_pybind.types import UINT8
```
UINT8 is present twice please check

fiona-gladwin commented on rocAL_pybind/amd/rocal/types.py:
```diff
@@ -137,6 +139,7 @@
     FLOAT: ("FLOAT", FLOAT),
     FLOAT16: ("FLOAT16", FLOAT16),
     UINT8: ("UINT8", UINT8),
```
UINT8 is present twice please check

fiona-gladwin commented on rocAL_pybind/rocal_pybind.cpp:
```diff
@@ -1024,6 +1034,10 @@ py::class_<rocalListOfTensorList>(m, "rocalListOfTensorList")
             py::return_value_policy::reference);
+    m.def("numpyReaderSource", &rocalNumpyFileSource, "Reads data from numpy files",
+          py::return_value_policy::reference);
+    m.def("numpyReaderSourceShard", &rocalNumpyFileSourceSingleShard, "Reads data from numpy files according to the shard id and number of shards",
```
Please retain only 1 API

fiona-gladwin commented on rocAL_pybind/rocal_pybind.cpp:
```diff
@@ -1024,6 +1034,10 @@ py::class_<rocalListOfTensorList>(m, "rocalListOfTensorList")
             py::return_value_policy::reference);
     m.def("audioDecoder", &rocalAudioFileSource, "Reads file from the source given and decodes it",
             py::return_value_policy::reference);
+    m.def("numpyReaderSource", &rocalNumpyFileSource, "Reads data from numpy files",
```
Can we just name it numpyReader for pybind?

fiona-gladwin commented on tests/python_api/numpy_reader.py:
```diff
@@ -0,0 +1,68 @@
+            os.makedirs(path)
+    except OSError as error:
+        print(error)
+    if (sys.argv[2] == "cpu"):
+        rocal_cpu = True
+    else:
+        rocal_cpu = False
```
```suggestion
    if (sys.argv[2] == "cpu"):
        rocal_cpu = True
    else:
        rocal_cpu = False
```
rocal_cpu = True if sys.argv[2] == "cpu" else False

SundarRajan28 committed:
Merge branch 'numpy_reader_cpp_changes' into numpy_reader_python_changes

SundarRajan28 committed:
Adding reader test for np reader

SundarRajan28 committed:
Resolving review comments

SundarRajan28 committed:
Merge branch 'develop' into numpy_reader_python_changes

SundarRajan28 committed:
Merge branch 'numpy_reader_cpp_changes' into numpy_reader_python_changes

kiritigowda committed:
Merge branch 'develop' into numpy_reader_python_changes

kiritigowda committed:
Merge branch 'develop' into numpy_reader_python_changes

SundarRajan28 committed:
Merge branch 'numpy_reader_cpp_changes' into numpy_reader_python_changes

LakshmiKumar23 committed:
Test - clean data paths (#298)

SundarRajan28 committed:
Merge branch 'numpy_reader_cpp_changes' into numpy_reader_python_changes

SundarRajan28 committed:
Resolving review comments

SundarRajan28 committed:
Macro changes

lucbruni-amd committed:
Test - Fix README inconsistencies (#299)

* Test - Fix README inconsistencies

* Test - Fix typo

* Update tests/cpp_api/unit_tests/README.md

Co-authored-by: spolifroni-amd <Sandra.Polifroni@amd.com>

---------

Co-authored-by: spolifroni-amd <Sandra.Polifroni@amd.com>

SundarRajan28 committed:
Merge branch 'numpy_reader_cpp_changes' into numpy_reader_python_changes

dependabot[bot] committed:
Docs - Bump rocm-docs-core[api_reference] from 1.17.1 to 1.18.1 in /docs/sphinx (#300)

Bumps [rocm-docs-core[api_reference]](https://github.com/ROCm/rocm-docs-core) from 1.17.1 to 1.18.1.
- [Release notes](https://github.com/ROCm/rocm-docs-core/releases)
- [Changelog](https://github.com/ROCm/rocm-docs-core/blob/develop/CHANGELOG.md)
- [Commits](https://github.com/ROCm/rocm-docs-core/compare/v1.17.1...v1.18.1)

---
updated-dependencies:
- dependency-name: rocm-docs-core[api_reference]
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

SundarRajan28 committed:
Deps update: Support turbojpeg package (#296)

* Removing tj3 API usage

* Adding turbojpeg package in rocAL setup script

* Update changelog

* Add turbojpeg rpm pkg

---------

Co-authored-by: SundarRajan98 <svaithiy@amd.com>
Co-authored-by: Lakshmi Kumar <lakshmi.kumar@amd.com>

kiritigowda committed:
Merge branch 'develop' into numpy_reader_python_changes
